src/charm_option.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+import charmhelpers.core.host as ch_host
+from typing import List, Union
+
+
+class CharmCephOption():
+    """Manage a charm option to ceph command to manage that option"""
+
+    def __init__(
+        self, charm_option_name, ceph_option_name, min_version=None
+    ):
+        self.charm_option_name = charm_option_name
+        self.ceph_option_name = ceph_option_name
+        self.min_version = min_version
+
+    def is_supported(self) -> bool:
+        """Is the option supported on this unit"""
+        if self.min_version:
+            return self.minimum_supported(self.min_version)
+        return True
+
+    def minimum_supported(self, supported_version: str) -> bool:
+        """Check if installed Ceph release is >= to supported_version"""
+        return ch_host.cmp_pkgrevno('ceph-common', supported_version) >= 0
+
+    def convert_option(self, value: Union[bool, str, int]) -> List[str]:
+        """Convert a value to the corresponding value part of the ceph
+            dashboard command"""
+        return [str(value)]
```

Comment:
I believe this method could be omitted since it's only used once and the logic is not so complex